### PR TITLE
feat: player names UI improvements

### PR DESCRIPTION
**Pull request by AjimenezDCL**
This PR creates optimizes the player names rendering as follow:

- Set the TMPro text in a material in the alpha tested queues (had to extend a TMPro shader to enable ZWrite.
- Create 2 extra materials for the background and the text when they are highlighted (over everything), with ZTest on `Always`.
- Profit.

## Test
Pay attention to player names, no significative change should be noticed. It should behave just as it does in production.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### <samp>🤖 Generated by Copilot at f9ba3bc</samp>

This pull request improves the rendering and visibility of the player name in the scene. It adds new shaders, materials, and prefabs that enable depth testing and writing for the player name text and background, and uses them to toggle the player name on and off. It also fixes some minor issues with the prefab and meta files.

### CODE REVIEW

**Review by kwsskulinski — APPROVED** [review]
No issues have been found.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/PlayerName/PlayerName.cs [hunk 1]
```diff
@@ -132,8 +135,11 @@ public void RemoveVisibilityConstaint(string constraint)
     public void SetForceShow(bool forceShow)
     {
         canvas.enabled = forceShow || namesVisible.Get();
-        canvas.sortingOrder = forceShow ? FORCE_CANVAS_SORTING_ORDER : DEFAULT_CANVAS_SORTING_ORDER;
+
+        //canvas.sortingOrder = forceShow ? FORCE_CANVAS_SORTING_ORDER : DEFAULT_CANVAS_SORTING_ORDER;
```
Comment (sandrade-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/PlayerName/PlayerName.cs:139]: Do we want to leave it commented for any reason? or we can remove it?

**Review by sandrade-dcl — APPROVED** [review]

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/PlayerName/PlayerName.cs (hunk 1 above)
Comment (AjimenezDCL) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/PlayerName/PlayerName.cs:139]: Nope, good catch

**Review by Kinerius — APPROVED** [review]
Nice

**Pull request merged**